artifacts/src/test/java/org/eclipse/dsp/context/fixtures/AbstractJsonLdTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ *  Copyright (c) 2024 Metaform Systems, Inc.
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Metaform Systems, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.dsp.context.fixtures;
+
+import com.apicatalog.jsonld.JsonLdError;
+import com.apicatalog.jsonld.JsonLdOptions;
+import com.apicatalog.jsonld.document.Document;
+import com.apicatalog.jsonld.document.JsonDocument;
+import com.apicatalog.jsonld.loader.DocumentLoader;
+import com.apicatalog.jsonld.loader.DocumentLoaderOptions;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsonp.JSONPModule;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SchemaLocation;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonStructure;
+import org.junit.jupiter.api.BeforeEach;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.apicatalog.jsonld.JsonLd.compact;
+import static com.apicatalog.jsonld.JsonLd.expand;
+import static org.eclipse.dsp.DspConstants.DSP_CONTEXT;
+import static org.eclipse.dsp.DspConstants.DSP_PREFIX;
+import static org.eclipse.dsp.DspConstants.ODRL_CONTEXT;
+import static com.networknt.schema.SpecVersion.VersionFlag.V202012;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Base class for Json-Ld expansion and compaction tests.
+ */
+public abstract class AbstractJsonLdTest {
+    protected ObjectMapper mapper;
+    protected JsonStructure compactionContext;
+    protected JsonLdOptions options;
+
+    private static final String CLASSPATH_SCHEMA = "classpath:/";
+    private static final String CONTEXT_REFERENCE = format("{\"@context\": [\"%s\"]}", DSP_CONTEXT);
+
+    protected void verifyRoundTrip(String jsonFile, String schemaFile) {
+        try {
+            var stream = getClass().getResourceAsStream(jsonFile);
+            var message = mapper.readValue(stream, JsonObject.class);
+            var expanded = expand(JsonDocument.of(message)).options(options).get();
+            var compacted = compact(JsonDocument.of(expanded), JsonDocument.of(compactionContext)).options(options).get();
+
+            var schemaFactory = JsonSchemaFactory.getInstance(V202012, builder ->
+                    builder.schemaMappers(schemaMappers -> schemaMappers.mapPrefix(DSP_PREFIX, CLASSPATH_SCHEMA))
+            );
+
+            var schema = schemaFactory.getSchema(SchemaLocation.of(DSP_PREFIX + schemaFile));
+            var result = schema.validate(mapper.convertValue(compacted, JsonNode.class));
+            assertThat(result.isEmpty()).isTrue();
```

Comment:
I'm not sure I understand. If someone adds a non-IRI property name, do you mean if they don't perform Json-Ld processing it will validate against the schema, but if Json-Ld processing is done and the context does not contain a corresponding term definition it will fail because it will be removed? In that case, maybe we need to add wording to the spec. Here's what is in [DCP](https://eclipse-dataspace-dcp.github.io/decentralized-claims-protocol/#schemas-contexts-and-message-processing):

> All protocol messages are normatively defined by a [json-schema]. This specification also uses JSON-LD 1.1 and provides a Json-Ld context to serialize data structures and message types as it facilitates extensibility. The Json-Ld context is designed to produce message serializations using compaction that validate against the Json Schema for the given message type. This allows implementations to choose whether to process messages as plain Json or as Json-Ld and maintain interoperability between those approaches. Extensions that use Json-Ld are encouraged to provide similar contexts that facilitate this approach to interoperability.

The last sentence implies that but it could be more explicit:

> If extensions do not provide a context that supports compaction of non-IRI property names, interoperability between implementations that do not perform Json-Ld processing may be impaired.

Perhaps we should create a separate issue for this?

Reply:
opened one https://github.com/eclipse-dataspace-protocol-base/DataspaceProtocol/issues/59